### ISSUE TITLE
Docker Desktop host networking conflicts with docker-mac-net-connect and causes login to fail

### ISSUE THREAD

**Issue by lhotari** [chipmk/docker-mac-net-connect#41]
Docker Desktop 4.34.0 ([release notes](https://docs.docker.com/desktop/release-notes/#4340)) comes with host networking support. This conflicts with docker-mac-net-connect. 

The problem digests as docker login failing. In the Docker Desktop UI, there's a popup visible for a few seconds where there's an error message about a port conflict.

The problem was resolved by running `sudo brew services stop chipmk/tap/docker-mac-net-connect` since I had enabled the service when I installed `docker-mac-net-connect`.

As a resolution to this issue, I guess it would be sufficient to mention this in the documentation of docker-mac-net-connect.

**Comment by kmahelona** [comment]
Does that mean you don't have to use docker-mac-net-connect to expose the containers' IPs to the host?

**Comment by lhotari** [comment]
> Does that mean you don't have to use docker-mac-net-connect to expose the containers' IPs to the host?

Do I have to use it? If I'd like to be able to access the container IPs, it would continue to be useful. The reported problem here is that docker-mac-net-connect conflicts with Docker Desktop's host network feature and it's hard to find out the reason.

**Comment by gregnr** [comment]
Thanks for reporting @lhotari and the follow up @RafalSkolasinski. To help me reproduce this, the port conflict error occurs if:

1. You enable host networking via **Settings** > **Resources** > **Network** > **Enable host networking**
2. You are running `docker-mac-net-connect`
3. You attempt to login to Docker via `docker login`?

**Comment by gregnr** [comment]
Update: I reproduced the port conflict error popup by simply enabling host networking mode and waiting a few seconds. For me though, the L3 `docker-mac-net-connect` tunnel continued to work and `docker login` succeeded. I tried doing this is both orders too:
1. Start `docker-mac-net-connect`, then enable host networking
2. Enable host networking, then start `docker-mac-net-connect`

Both produced the port conflict error, but neither caused any tunnel or `docker login` issues.

If there is an extra step I'm missing please let me know. If the problem is simply the warning, this is probably less critical than I first thought, though we should still try to find a way to prevent this. I'm suspicious it has to do with the WireGuard kernel module running in the VM making an outbound UDP connection to the macOS host, since Docker for Mac's host networking solution seems to do some sort of port mapping/mirroring between the VM and the macOS host.

**Comment by RafalSkolasinski** [comment]
Hi @gregnr, for me it was the `1. Start docker-mac-net-connect, then enable host networking` that did it, where enabling host networking was restarting docker desktop. 

I would say that running `docker login` is just a workaround and the problem being that docker desktop logged out and the "Sign in" feature through the UI was completely broken.

---

I am also observing recently that some containers through started are not appearing in `docker ps` and docker desktop but I cannot say that this is related here...

**Comment by gregnr** [comment]
Thanks for the details. What broke with the UI sign in? I tried reproducing by:

1. Start `docker-mac-net-connect`
2. Enable host networking
3. Click "Sign in" at top right in Docker Desktop UI

After redirecting to the browser and signing in, I was taken back to the desktop UI which had logged in successfully.

**Comment by gregnr** [comment]
To confirm, on 3 after you are redirected back, Docker Desktop remains logged out for you instead of logged in? When I ran the above steps I logged in successfully. I'm not sure I understand what the problem is.

**Comment by RafalSkolasinski** [comment]
Yes, it does stay logged out until I do `docker login` from the terminal. I am surprised it is not happening for you. 

I reproduced it on two independent machines. It also happens later on if with both enabled I just restart docker desktop. 
It is very consistent with having both host networking and `docker-mac-net-connect` on at the same time.
